pkgs/top-level/all-packages.nix

```diff
@@ -2141,6 +2143,8 @@ in
     gcc = gcc49; # doesn't build with gcc5
   };
 
+  jp = (callPackage ../development/tools/jp { }).bin // { outputs = [ "bin" ]; };
```

Comment:
I didn't quite follow at first either, but I looked at a few of the other Go apps and copied the behavior. Crawling through nixpkgs looking at how Go packages were created, I found https://nixos.org/releases/nixpkgs/nixpkgs-16.09pre86545.65ac26e/manual/#ex-buildGoPackage which mentions how it generates multiple outputs. Running a `nix-build . -A jp.all` or `jp.bin` got what I wanted but thats not the normal experience I expect with installing packages.

Maybe it could be spelled out more thoroughly in the docs? There's just a bit more going on behind the scenes with buildGoPackage. For this one, I'm only interested in the bin/ directory and want to discard the rest of the output as there's nothing useful in there.

Also, fwiw, this is building on my mac but it did take a good long time for the compilation to finish, travis appears to be building a lot and it didn't finish.